sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala

```diff
@@ -33,15 +36,34 @@ sealed trait IdentifierWithDatabase {
    */
   private def quoteIdentifier(name: String): String = name.replace("`", "``")
 
-  def quotedString: String = {
-    val replacedId = quoteIdentifier(identifier)
-    val replacedDb = database.map(quoteIdentifier(_))
+  def quotedString: String = quotedString(None)
+  def quotedString(catalog: String): String = quotedString(Some(catalog))
 
-    if (replacedDb.isDefined) s"`${replacedDb.get}`.`$replacedId`" else s"`$replacedId`"
+  def quotedString(catalog: Option[String]): String = {
```

Comment:
it is hacky that we access the conf inside identifier that it may appear in anywhere. I make a new pr which pull out the config see https://github.com/apache/spark/pull/37021